cornac/models/narre/recom_narre.py

```diff
@@ -0,0 +1,296 @@
+
+    n_factors: int, default: 32
+        The dimension of the user/item's latent factors.
+
+    attention_size: int, default: 16
+        Attention size
+
+    kernel_sizes: list, default: [3]
+        List of kernel sizes of conv2d
+
+    n_filters: int, default: 64
+        Number of filters
+
+    dropout_rate: float, default: 0.5
+        Dropout rate of neural network dense layers
+
+    max_text_length: int, default: 50
+        Maximum number of tokens in a review instance
+
+    batch_size: int, default: 64
+        Batch size
+
+    max_iter: int, default: 10
+        Max number of training epochs
+
+    trainable: boolean, optional, default: True
+        When False, the model will not be re-trained, and input of pre-trained parameters are required.
+
+    verbose: boolean, optional, default: True
+        When True, running logs are displayed.
+
+    init_params: dictionary, optional, default: None
+        Initial parameters, pretrained_word_embeddings could be initialized here, e.g., init_params={'pretrained_word_embeddings': pretrained_word_embeddings}
+
+    seed: int, optional, default: None
+        Random seed for weight initialization.
+        If specified, training will take longer because of single-thread (no parallelization).
+
+    References
+    ----------
+    * Chen, C., Zhang, M., Liu, Y., & Ma, S. (2018, April). Neural attentional rating regression with review-level explanations. In Proceedings of the 2018 World Wide Web Conference (pp. 1583-1592).
+    """
+
+    def __init__(
+        self,
+        name="NARRE",
+        embedding_size=100,
+        id_embedding_size=32,
+        n_factors=32,
+        attention_size=16,
+        kernel_sizes=[3],
+        n_filters=64,
+        dropout_rate=0.5,
+        max_text_length=50,
+        batch_size=64,
+        max_iter=10,
+        trainable=True,
+        verbose=True,
+        init_params=None,
+        seed=None,
+    ):
+        super().__init__(name=name, trainable=trainable, verbose=verbose)
+        self.seed = seed
+        self.embedding_size = embedding_size
+        self.id_embedding_size = id_embedding_size
+        self.n_factors = n_factors
+        self.attention_size = attention_size
+        self.n_filters = n_filters
+        self.kernel_sizes = kernel_sizes
+        self.dropout_rate = dropout_rate
+        self.max_text_length = max_text_length
+        self.batch_size = batch_size
+        self.max_iter = max_iter
+        # Init params if provided
+        self.init_params = {} if init_params is None else init_params
+
+    def _init(self):
+        import tensorflow as tf
+        if self.seed is not None:
+            self.rng = get_rng(self.seed)
+            tf.random.set_seed(self.seed)
+        self.n_users, self.n_items = self.train_set.num_users, self.train_set.num_items
+        self.n_vocab = self.train_set.review_text.vocab.size
+        self.pretrained_word_embeddings = self.init_params.get('pretrained_word_embeddings')
```

Comment:
This can be passed directly to the constructor of Model. It doesn't have to be an attribute of rec object as it might significantly increase the size of the saved checkpoint.